hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -214,8 +214,9 @@ public void monitorDirAndForwardSplits(SourceContext<MergeOnReadInputSplit> cont
     }
     IncrementalInputSplits.Result result =
         incrementalInputSplits.inputSplits(metaClient, this.issuedOffset, this.cdcEnabled);
-    if (result.isEmpty()) {
+    if (result.equals(IncrementalInputSplits.Result.EMPTY)) {
```

Comment:
should we always update the `issuedInstant` if it is not null?

Reply:
If value of `read.speed.limit` is smaller than number of empty instants  ,  result.isEmpty is true and issuedInstant is not updated.

Reply:
it is not updated but can still be used to reset the variable.

Reply:
In speed limit scenario, it can not be updated any more.  I try to give an example to explain this:

assume current active timeline is instant1、instant2、instant3、instant4、instant5、instant6 ...
If the issuedInstant is instant2 and read.speed.limit=2 and instant3 and instant4 are empty commits,  then the inputSplits method return `Result.instance(inputSplits, endInstant, offsetToIssue)`.
In this return, inputSplits is Collections.emptyList() , endInstant is  instant4,  offsetToIssue is max(instant3 completionTime, instant4.completionTime).

So result.isEmpty() is true and monitorDirAndForwardSplits() returned, but offsetToIssue and issuedInstant are not updated. Next time when monitorDirAndForwardSplits() is called,  the inputSplits method return  the same as above, endless loop.
@danny0405

Reply:
I meant just to use `instant4.completionTime` to update the `issuedInstant` in this case.

Reply:
This bug is easy to be reproduced. Let us take UT `TestStreamReadMonitoringFunction#testConsumeForSpeedLimitWhenEmptyCommitExists()` as an example:
Step1: create 4 empty commit
Step2: trigger streaming read from first instant and set READ_COMMITS_LIMIT 2
Step3: assert current IssuedOffset couldn't be null. 
Base on `IncrementalInputSplits#inputSplits => .startCompletionTime(issuedOffset != null ? issuedOffset : this.conf.getString(FlinkOptions.READ_START_COMMIT))`
If IssuedOffset still was null, hudi would take FlinkOptions.READ_START_COMMIT again, which means streaming read is blocked.

Without this PR this UT will be failed which means streaming reading empty commits will be blocked.

Reply:
We only update the `issuedOffset` with non-null `issuedInstant`, does it make sense?

Reply:
make sense.  PTAL